debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/connection/PostgresConnectionIT.java

```diff
@@ -111,6 +114,32 @@ public void shouldDropReplicationSlot() throws Exception {
         }
     }
 
+    @Test
+    public void temporaryReplicationSlotsShouldGetDroppedAutomatically() throws Exception {
+        try(ReplicationConnection replicationConnection = TestHelper.createForReplication("test", true)) {
+            PgConnection pgConnection = getUnderlyingConnection(replicationConnection);
+
+            // temporary replication slots are not supported by Postgres < 10
+            if (pgConnection.getServerMajorVersion() < 10) {
+                return;
+            }
+
+            // simulate ungraceful shutdown by closing underlying database connection
+            pgConnection.close();
+
+            try (PostgresConnection connection = TestHelper.create()) {
+                assertFalse("postgres did not drop replication slot", connection.dropReplicationSlot("test"));
+            }
+        }
+    }
+
+    private PgConnection getUnderlyingConnection(ReplicationConnection connection) throws Exception {
+        Field connField = JdbcConnection.class.getDeclaredField("conn");
```

Comment:
Interesting way of testing :) This might bite us when moving to the Java Module System later on, but ok with me for the time being.